caravel/views.py

```diff
@@ -1588,19 +1588,53 @@ def activity_per_day(self):
 
     @api
     @has_access_api
-    @expose("/tables/<db_id>/<schema>")
+    @expose("/schemas/<db_id>")
+    @cached_view(timeout=600)
+    def schemas(self, db_id):
+        # db_id = request.args.get('db_id')
+        database = (
+            db.session
+            .query(models.Database)
+            .filter_by(id=db_id)
+            .one()
+        )
+        return Response(
+            json.dumps({'schemas': database.all_schema_names()}),
+            mimetype="application/json")
+
+    @api
+    @has_access_api
+    @expose("/tables/<db_id>/<schema>/")
+    @cached_view(timeout=600)
```

Comment:
do we want to cache if the schema is defined, that "up to" 10 minute lag is bad if you just CREATE TABLE as and want to work off of your summary table...